docs/docs/addons-update.md

```diff
@@ -19,7 +19,7 @@ var update = require('react-addons-update'); // ES5 with npm
 
 ## Overview
 
-React lets you use whatever style of data management you want, including mutation. However, if you can use immutable data in performance-critical parts of your application it's easy to implement a fast [`shouldComponentUpdate()`](/react/docs/react-component.html#shouldcomponentupdate) method to significantly speed up your app.
+React lets you use whatever style of data management you want, including mutation. However, React's state should be treated as immutable. Immutable data makes it easy to implement a fast [`shouldComponentUpdate()`](/react/docs/react-component.html#shouldcomponentupdate) method to significantly speed up your app.
```

Comment:
I'm worried this might be too specific since the original sentence covered immutable data in props and context as well.